### ISSUE TITLE
[ENHANCEMENT]:Support for new xtopics in K- and L-series

### ISSUE THREAD

**Issue by ZonPV** [edterbak/NodeRed_Heishamon_control#150]
At the moment there are some new topics like Heat_Power_Consumption and DHW_Power_Consumption available in the latest Heishamon firmware 3.2.3 that give no results in the dashboard of version 24.2. So the graphic Energy stays empty.

Also there is no value according defrosts. But that is because there is no support for that in the actual Heishamon firmware. Igor is busy with that subject. Maybe I have to split these issues?

Should be nice if there is time to solve these issues.

Als reminder: https://gathering.tweakers.net/forum/list_message/77515844#77515844

**Comment by edterbak** [comment]
I thought I had.
Can you give me the exact topics, gathered from MQTT explorer?

I presume these are incorrect then:

_-Heat_Power_Consumption_Extra vs Heat_Power_Consumption
-Cool_Power_Consumption_Extra vs Cool_Power_Consumption
-DHW_Power_Consumption_Extra vs DHW_Power_Consumption
-Heat_Power_Production_Extra vs Heat_Power_Production
-Cool_Power_Production_Extra vs Cool_Power_Production
-DHW_Power_Production_Extra vs DHW_Power_Production_

**Comment by ZonPV** [comment]
Sorry for delay. I expected a mail from Github in case of progress...

This is the complete output of MQTT Explorer:
main
Heatpump_State = 1
Pump_Flow = 20.18
Force_DHW_State = 0
Quiet_Mode_Schedule = 0
Operating_Mode_State = 0
Main_Inlet_Temp = 24
Main_Outlet_Temp = 26.50
Main_Target_Temp = 32
Compressor_Freq = 56
DHW_Target_Temp = 52
DHW_Temp = 44
Operations_Hours = 317
Operations_Counter = 164
Main_Schedule_State = 0
Outside_Temp = -1
Heat_Power_Production = -200
Heat_Power_Consumption = -200
Powerful_Mode_Time = 0
Quiet_Mode_Level = 0
Holiday_Mode_State = 0
ThreeWay_Valve_State = 0
Outside_Pipe_Temp = -5
DHW_Heat_Delta = -4
Heat_Delta = 5
Cool_Delta = 5
DHW_Holiday_Shift_Temp = -15
Defrosting_State = 0
Z1_Heat_Request_Temp = 22
Z1_Cool_Request_Temp = 28
Z1_Heat_Curve_Target_High_Temp = 32
Z1_Heat_Curve_Target_Low_Temp = 35
Z1_Heat_Curve_Outside_High_Temp = 15
Z1_Heat_Curve_Outside_Low_Temp = -5
Room_Thermostat_Temp = 20
Z2_Heat_Request_Temp = 35
Z2_Cool_Request_Temp = 0
Z1_Water_Temp = -78
Z2_Water_Temp = -78
Cool_Power_Production = -200
Cool_Power_Consumption = -200
DHW_Power_Production = -200
DHW_Power_Consumption = -200
Z1_Water_Target_Temp = 32
Z2_Water_Target_Temp = 35
Error = No error
Room_Holiday_Shift_Temp = -15
Buffer_Temp = -78
Solar_Temp = -78
Pool_Temp = -78
Main_Hex_Outlet_Temp = 26
Discharge_Temp = 32
Inside_Pipe_Temp = 23
Defrost_Temp = -128
Eva_Outlet_Temp = -2
Bypass_Outlet_Temp = -128
Ipm_Temp = -128
Z1_Temp = 20
Z2_Temp = -128
DHW_Heater_State = 1
Room_Heater_State = 0
Internal_Heater_State = 0
External_Heater_State = 0
Fan1_Motor_Speed = 640
Fan2_Motor_Speed = 0
High_Pressure = 18.6
Pump_Speed = 2400
Low_Pressure = 0
Compressor_Current = 5.8
Force_Heater_State = 0
Sterilization_State = 0
Sterilization_Temp = 63
Sterilization_Max_Time = 15
Z1_Cool_Curve_Target_High_Temp = 15
Z1_Cool_Curve_Target_Low_Temp = 10
Z1_Cool_Curve_Outside_High_Temp = 30
Z1_Cool_Curve_Outside_Low_Temp = 20
Heating_Mode = 1
Heating_Off_Outdoor_Temp = 24
Heater_On_Outdoor_Temp = 0
Heat_To_Cool_Temp = 15
Cool_To_Heat_Temp = 10
Cooling_Mode = 0
Z2_Heat_Curve_Target_High_Temp = 35
Z2_Heat_Curve_Target_Low_Temp = 35
Z2_Heat_Curve_Outside_High_Temp = 15
Z2_Heat_Curve_Outside_Low_Temp = -5
Z2_Cool_Curve_Target_High_Temp = 15
Z2_Cool_Curve_Target_Low_Temp = 10
Z2_Cool_Curve_Outside_High_Temp = 30
Z2_Cool_Curve_Outside_Low_Temp = 20
Room_Heater_Operations_Hours = 0
DHW_Heater_Operations_Hours = 0
Heat_Pump_Model = 255
Pump_Duty = 95
Zones_State = 0
Max_Pump_Duty = 150
Heater_Delay_Time = 30
Heater_Start_Delta = -4
Heater_Stop_Delta = -2
Buffer_Installed = 0
DHW_Installed = 1
Solar_Mode = 0
Solar_On_Delta = 10
Solar_Off_Delta = 5
Solar_Frost_Protection = 5
Solar_High_Limit = 80
Pump_Flowrate_Mode = 0
Liquid_Type = 0
Alt_External_Sensor = 0
Anti_Freeze_Mode = 1
Optional_PCB = 0
Z1_Sensor_Settings = 2
Z2_Sensor_Settings = 0
Buffer_Tank_Delta = 5
External_Pad_Heater = 0
extra
Heat_Power_Consumption_Extra = 1378
Cool_Power_Consumption_Extra = 0
DHW_Power_Consumption_Extra = 0
Heat_Power_Production_Extra = 4031
Cool_Power_Production_Extra = 0
DHW_Power_Production_Extra = 0
1wire
28c96e7204000095 = 

A screenshot:

[https://ibb.co/8bF1NLG](url)

**Comment by ZonPV** [comment]
Als ik bovenstaande lijst doorneem mis ik nog wel meer waardes:

Room_Heater_Operations_Hours = 0
DHW_Heater_Operations_Hours = 0

En lijken volgende waardes verwisseld:

DHW_Heater_State = 1
Room_Heater_State = 0

Op bedienpaneel staat momenteel alleen icoon van zonnetje met daarbij de aanduiding dat de buitenunit actief is.
DHW wordt alleen in de ochtend gemaakt. Via het interne programma schakelt de wp even naar mixed mode.. Na 12:00 uur weer terug naar alleen ruimteverwarming.

Ik moet daarbij aangeven dat ik momenteel de Heishamon niet in bedrijf heb. Alleen voor je bovenstaande vraag even de Heishamon aangekoppeld. Het bedienpaneel als thermostaat in gebruik in de huiskamer i.v.m. WAF..

**Comment by edterbak** [comment]
```
extra
Heat_Power_Consumption_Extra = 1378
Cool_Power_Consumption_Extra = 0
DHW_Power_Consumption_Extra = 0
Heat_Power_Production_Extra = 4031
Cool_Power_Production_Extra = 0
DHW_Power_Production_Extra = 0
```

I asume that means this:


```
panasonic_heat_pump/extra/Heat_Power_Consumption_Extra
panasonic_heat_pump/extra/Cool_Power_Consumption_Extra
panasonic_heat_pump/extra/DHW_Power_Consumption_Extra
panasonic_heat_pump/extra/Heat_Power_Production_Extra
panasonic_heat_pump/extra/Cool_Power_Production_Extra
panasonic_heat_pump/extra/DHW_Power_Production_Extra
```

I had the topics under main. Thats probably why they are not found. 😃

**Comment by ZonPV** [comment]
There is a screenshot from MQTT Explorer in my comment of Thursday:

![MQTT Panasonic](https://github.com/edterbak/NodeRed_Heishamon_control/assets/11488827/0ef83113-6387-4bcf-a972-b85514adae30)

**Comment by edterbak** [comment]
Update has been released, v24.03.
Can you confirm it is working fine now?

**Comment by ZonPV** [comment]
I can confirm that those ExtraTopics work fine in v 24.03. Thx for all your effort
